xarray/core/dataset.py

```diff
@@ -2018,13 +2019,15 @@ def func(self, *args, **kwargs):
         return func
 
     @staticmethod
-    def _binary_op(f, reflexive=False, join='inner', fillna=False):
+    def _binary_op(f, reflexive=False, join=None, fillna=False):
         @functools.wraps(f)
         def func(self, other):
             if isinstance(other, groupby.GroupBy):
                 return NotImplemented
             if hasattr(other, 'indexes'):
-                self, other = align(self, other, join=join, copy=False)
+                # if user does not specify join, default to OPTIONS['join']
+                how_to_join = join if join is not None else OPTIONS['join']
```

Comment:
Out of curiosity, why create a new variable `how_to_join`? Convention in [the only other use of `OPTIONS`](https://github.com/pydata/xarray/blob/f40d323c76f6cab9615567e4ba7afde7860f7775/xarray/core/formatting.py#L191-L192) is:

    if max_width is None:
        max_width = OPTIONS['display_width']